### PR TITLE
refactor: centralize billing cycle window resolution

### PR DESCRIPTION
**Pull request by QinyingShi**
## Summary
- add a small billing cycle transition helper for order effective window resolution
- keep existing subscription wrapper functions and behavior intact
- make billing overview return the current wallet balance from bucket state so stale wallet snapshots do not hide newly granted credits
- show `0` credits in the admin sidebar instead of the placeholder when the loaded balance is zero
- freeze credit pack buckets in operation reservations when no active plan exists, while keeping held credit pack balances visible in billing details
- clarify the frozen credit-pack prompt and tighten review-requested regressions for balance rendering and plan reactivation
- keep overview credit status/debug permission aligned with the same consumable balance projection and keep owned credit packs visible after old consumption windows end

## Scope
- Billing-R3 first step: cycle-window parsing plus the dev02 balance-display and frozen credit-pack regressions found during validation
- no grant activation changes
- no bucket expiration changes
- no subscription status transition changes
- no repair, audit, schema, or migration changes

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved billing cycle effective start/end boundary handling for renewals, top-ups, and subscription starts, with safer fallbacks when data is missing.
  - Billing overview now recalculates available/reserved credits from the current wallet snapshot and derives limit state based on current consumable balances.
  - Operation credit reservations now correctly deny top-ups when there’s no active effective top-up subscription or when outside the allowed consumption window.
  - Billing UI now consistently shows `0` available credits and properly reflects held/top-up credits.
- **Refactor**
  - Centralized cycle effective boundary logic for reuse across billing flows.
- **Localization**
  - Updated `subscriptionInactive` billing message copy (EN/FR/ZH).
- **Tests**
  - Added/expanded unit and UI tests for cycle transitions, overview/sidebar rendering, and reservation edge cases.

### CODE REVIEW

File: src/api/tests/service/billing/test_operation_credit_reservations.py [hunk 1]
```diff
@@ -295,6 +299,79 @@ def test_reserve_operation_credits_rejects_insufficient_balance(
...
+        assert bucket.reserved_credits == Decimal("0E-10")
+        assert (
+            CreditLedgerEntry.query.filter_by(
+                creator_bid=creator_bid,
+                entry_type=CREDIT_LEDGER_ENTRY_TYPE_HOLD,
+            ).count()
+            == 0
+        )
+
+
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/tests/service/billing/test_operation_credit_reservations.py:411]: ✅ Addressed in commit b338467: extended the frozen top-up reservation test to reactivate the same subscription, retry reservation, and assert HOLD plus wallet/bucket balances.

File: src/api/flaskr/service/billing/read_models.py [hunk 2]
```diff
@@ -603,8 +606,20 @@ def build_billing_overview(
...
+                    snapshot_at=now_utc(),
+                )
+            )
+            wallet_payload.available_credits = credit_decimal_to_number(
+                available_credits
+            )
+            wallet_payload.reserved_credits = credit_decimal_to_number(reserved_credits)
         subscription_payload = _serialize_subscription(app, subscription)
-        limit_state = resolve_creator_limit_state(app, normalized_creator_bid)
+        limit_state = build_creator_limit_state_for_available_credits(available_credits)
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/service/billing/read_models.py]: ✅ Addressed in commit 27ebfb68: restored the billing-disabled overview guard so disabled billing keeps normal/debug-allowed instead of applying hardlimit from the projected balance.

File: src/api/tests/service/billing/test_operation_credit_reservations.py [hunk 3]
```diff
@@ -295,6 +299,178 @@ def test_reserve_operation_credits_rejects_insufficient_balance(
...
+                creator_bid=creator_bid,
+                entry_type=CREDIT_LEDGER_ENTRY_TYPE_HOLD,
+            ).count()
+            == 0
+        )
+        subscription = BillingSubscription.query.filter_by(
+            subscription_bid=f"subscription-{creator_bid}"
+        ).one()
+        subscription.current_period_start_at = datetime(2026, 1, 1, 0, 0, 0)
+        subscription.current_period_end_at = datetime(2099, 1, 1, 0, 0, 0)
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/tests/service/billing/test_operation_credit_reservations.py:382]: ✅ Addressed in commit 27ebfb68: patched the affected operation reservation tests to freeze now_utc at 2026-01-16 so expired subscription/window expectations remain deterministic.
